documentation/tutorials/get-started-with-ash-and-phoenix.md

```diff
@@ -0,0 +1,501 @@
+
+```elixir
+defmodule MyAshPhoenixAppWeb.ExampleLiveView do
+  use MyAshPhoenixAppWeb, :live_view
+  import Phoenix.HTML.Form
+  alias MyAshPhoenixApp.Blog.Post
+
+  def render(assigns) do
+    ~H"""
+    <h2>Posts</h2>
+    <div>
+    <%= for post <- @posts do %>
+      <div>
+        <div><%= post.title %></div>
+        <div><%= if Map.get(post, :content), do: post.content, else: "" %></div>
+        <button phx-click="delete_post" phx-value-post-id={post.id}>delete</button>
+      </div>
+    <% end %>
+    </div>
+    <h2>Create Post</h2>
+    <.form let={f} for={@create_form} phx-submit="create_post">
+      <%= text_input f, :title, placeholder: "input title" %>
+      <%= submit "create" %>
+    </.form>
+    <h2>Update Post</h2>
+    <.form let={f} for={@update_form} phx-submit="update_post">
+      <%= label f, :"post name" %>
+      <%= select f, :post_id, @post_selector %>
+      <%= text_input f, :content, value: "", placeholder: "input content" %>
+      <%= submit "update" %>
+    </.form>
+    """
+  end
+
+  def mount(_params, _session, socket) do
+    posts = Post.read_all!()
+
+    socket =
+      assign(socket,
+        posts: posts,
+        post_selector: post_selector(posts),
+        create_form: AshPhoenix.Form.for_create(Post, :create),
+        update_form: AshPhoenix.Form.for_update(List.first(posts, %Post{}), :update)
+      )
+
+    {:ok, socket}
+  end
+
+  def handle_event("delete_post", %{"post-id" => post_id}, socket) do
+    post_id |> Post.get_by_id!() |> Post.destroy!()
+    posts = Post.read_all!()
+
+    {:noreply, assign(socket, posts: posts, post_selector: post_selector(posts))}
+  end
+
+  def handle_event("create_post", %{"form" => %{"title" => title}}, socket) do
+    Post.create(%{title: title})
+    posts = Post.read_all!()
+
+    {:noreply, assign(socket, posts: posts, post_selector: post_selector(posts))}
+  end
+
+  def handle_event("update_post", %{"form" => form_params}, socket) do
+    %{"post_id" => post_id, "content" => content} = form_params
+
+    post_id |> Post.get_by_id!() |> Post.update!(%{content: content})
+    posts = Post.read_all!()
+
+    {:noreply, assign(socket, :posts, posts)}
+  end
+
+  defp post_selector(posts) do
+    for post <- posts do
+      {:"#{post.title}", post.id}
+    end
+  end
+end
+```
+
+You can see how using functions created by our `code_interface` makes it easy to integrate Ash with Phoenix.
+
+You may also notice this is the first time we've used the AshPhoenix library. The AshPhoenix library contains utilities to help Ash integrate with Phoenix and LiveView Seamlessly. One of these utilities is `AshPhoenix.Form` which can automatically produce changesets to be used in the forms.
+
+That's it for this guide. We've gone from 0 to a fully working Phoenix App using Ash. But we are really just scratching the surface of what can be done in Ash. Look below for what to look at next.
```

Comment:
It might be good to have a repo with a working  version of the app? That would help tyre-kickers get up to speed super quickly.

Reply:
https://github.com/team-alembic/my_ash_phoenix_project
Already got one :)

Reply:
linked it: [675a3a4](https://github.com/ash-project/ash/pull/504/commits/675a3a41730b743fd0bd8b5db419557a0068ff22)